src/Adapter/Invoice/InvoiceOptionsConfiguration.php

```diff
@@ -27,44 +27,60 @@
 namespace PrestaShop\PrestaShop\Adapter\Invoice;
 
 use PrestaShop\PrestaShop\Adapter\Configuration;
-use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Adapter\Shop\Context;
+use PrestaShop\PrestaShop\Core\Configuration\AbstractMultistoreConfiguration;
+use PrestaShop\PrestaShop\Core\Feature\FeatureInterface;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * Class InvoiceOptionsConfiguration is responsible for saving & loading invoice configuration.
  */
-final class InvoiceOptionsConfiguration implements DataConfigurationInterface
+final class InvoiceOptionsConfiguration extends AbstractMultistoreConfiguration
 {
     /**
-     * @var Configuration
+     * @var FormChoiceProviderInterface
      */
-    private $configuration;
+    private $invoiceModelByNameChoiceProvider;
 
     /**
+     * AbstractMultistoreConfiguration constructor.
+     *
      * @param Configuration $configuration
+     * @param Context $shopContext
+     * @param FeatureInterface $multistoreFeature
+     * @param FormChoiceProviderInterface $invoiceModelByNameChoiceProvider
      */
-    public function __construct(Configuration $configuration)
-    {
-        $this->configuration = $configuration;
+    public function __construct(
+        Configuration $configuration,
```

Comment:
I admit, I took parent constructor (`AbstractMultistoreConfiguration::__construct(Configuration $configuration, Context $shopContext, FeatureInterface $multistoreFeature)`) without thinking more :thinking:.

Won't it break something if child requires `ConfigurationInterface` and give `Configuration` to parent ? :thinking: 

I'll check this ^^

Reply:
[It looks it works](https://onlinephp.io/c/4f587) if we give the class called by parent constructor but if you use another one (allowed by the interface), it will throw a fatal error when calling parent class constructor.

Reply:
I don't think it would be a good idea to push this tricky behavior in the code and IMHO, refactoring abstract parent class is completely out of the scope of this PR.